### DISCUSSION TITLE
Welcome to dash-mpd-cli Discussions!

### DISCUSSION THREAD

**Discussion by emarsden** [emarsden/dash-mpd-cli#87] [General]
## 👋 Welcome!

We’re using Discussions as a place to connect with other dash-mpd-cli users, ask questions and share ideas. This is probably a better place to ask a question about the usage of dash-mpd-cli than the "Issues" tab.

**Comment by Crazyvod00** [comment]
Hi emardsen! I'm trying to use your cli, first of all, great idea! Do you think is possible to have the cli as GUI, to be more user-frienldly?

I thought, maybe the GUI could work in this way too:

1. Add dash manifest url or file
2. The mpd is shown with the chance to move the adaptation set in a different place and the chance to disable one or more adaption sets.
3. Create a rule that can be remembered with new links.
4. Chance to edit the base url, maybe with regular expressions too.
5. Chance to add a label.
6. Save the mpd as new one.

Cheers!

Crazy

**Reply by emarsden** [reply]
That might be useful to some people, but I'm not currently very interested in developing a GUI. If anyone is interested in that, the Rust crate [dash-mpd](https://github.com/emarsden/dash-mpd-rs), for which dash-mpd-cli is a CLI wrapper, might be useful.

**Comment by outhud** [comment]
Hi,

Have you some advice to offer on an issue I'm seeing with a multi-period mpd file with vtt subtitles? There are 10 periods, and I'm selecting 3 for download (2, 5 and 8)

The subtitles download as one file for each period. The contents of the three .vtt files are identical, and they're the subtitles for the entire video (all 10 periods)

x@x-hpenvy:~$ ls -l OutputTwo*
-rw-r--r-- 1 x x 59561 Jan 17 23:11 OutputTwo-p2.vtt
-rw-r--r-- 1 x x 59561 Jan 17 23:11 OutputTwo-p5.vtt
-rw-r--r-- 1 x x 59561 Jan 17 23:11 OutputTwo-p8.vtt

I see the same whether or not I use --no-period-concatenation

The issue is that the timestamps in the .vtt file are in sync with the concatenated audio/video only for the case where all 10 periods are downloaded. When only some periods are downloaded, the timestamps in the vtt won't be updated, so the subs go out of sync.

Can you advise on a good way I can handle cases like this?

(some other info: I looked at the .mpd file, I see the same full .vtt file referenced 10 times in it, once for each period)

**Reply by emarsden** [reply]
This is an interesting but tricky problem. The media player that retrieves the MPD knows the start time of each Period, so can adjust the subtitles for each Period. We currently lose this information when we save each Period as a separate media file. I will take a look at whether there are solutions (which will probably be specific to certain containers such as mkv or mp4) to this problem.

**Reply by outhud** [reply]
Thank you. 

I had started to think about a solution using ffmpeg to postprocess the .vtt file afterwards, seeing as the lengths of the skipped periods are known from the mpd, and ffmpeg can handle trimming and appending .vtt subtitle files using the normal -ss, -to and concat filter (including updating the timestamps)

**Comment by druh90cyber** [comment]
Hello, not posting this in issues becuase it could well be a problem on my end. Whenever I try to download anything it gives me this message:

[0s] [>-------------------------------------------------] Fetching DASH manifest                                        14:20:45  INFO Only simulating media downloads
14:20:45  INFO Fetching the DASH manifest
14:20:46 ERROR \x1b[1;31mDownload failed\x1b[0m: parse error "parsing DASH XML: Parsing(\n    \"XML parsing: DtdUnsupported(\\n    Span {\\n        start: 0,\\n        end: 15,\\n    },\\n)\",\n)"
PS C:\Users\helen\Videos\Transformers Armada\dash>

I use windows 11 and have installed dash-md-cli on my computer, the above command was executed in powershell. I quadruple-checked so I'm pretty sure I installed all the necessary dependecies but there might still be something I'm missing. Could I please have some help with this?

**Reply by emarsden** [reply]
It looks like the URL you are trying to fetch from isn't a standard DASH manifest. Can you share the URL? 

Here is an example of a DASH manifest: https://cloudflarestream.com/31c9291ab41fac05471db4e73aa11717/manifest/video.mpd

**Reply by druh90cyber** [reply]
There it is: https://www.youtube.com/watch?v=3hws2x_aJVU
I've also tried to downlad from netflix links to no avail.

**Reply by druh90cyber** [reply]
dash-mp-cli does seem to like the link you gave a lot more

**Reply by emarsden** [reply]
You'll need to use another tool such as yt-dlp to download from Youtube, but please note that it's against their terms of service.
